lightning/src/ln/functional_test_utils.rs

```diff
@@ -1418,14 +1418,23 @@ macro_rules! check_closed_broadcast {
 }
 
 /// Check that a channel's closing channel events has been issued
-pub fn check_closed_event(node: &Node, events_count: usize, expected_reason: ClosureReason, is_check_discard_funding: bool) {
+pub fn check_closed_event(node: &Node, events_count: usize, expected_reason: ClosureReason, is_check_discard_funding: bool,
+	expected_counterparty_node_ids: &[PublicKey], expected_channel_capacity: u64) {
 	let events = node.node.get_and_clear_pending_events();
 	assert_eq!(events.len(), events_count, "{:?}", events);
 	let mut issues_discard_funding = false;
-	for event in events {
+	for (idx, event) in events.into_iter().enumerate() {
 		match event {
-			Event::ChannelClosed { ref reason, .. } => {
+			Event::ChannelClosed { ref reason, counterparty_node_id, 
+				channel_capacity_sats, .. } => {
 				assert_eq!(*reason, expected_reason);
+				if let Some(counterparty_node_id) = counterparty_node_id {
+					assert_eq!(counterparty_node_id, expected_counterparty_node_ids[idx]);
+				}
+				
+				if let Some(channel_capacity) = channel_capacity_sats {
+					assert_eq!(channel_capacity, expected_channel_capacity);
+				}
```

Comment:
Since we're always setting this from this PR onward maybe it's better to unwrap here in tests so that it doesn't silently allow `None`s coming through?

```suggestion
				assert_eq!(counterparty_node_id.unwrap(), expected_counterparty_node_ids[idx]);
				assert_eq!(channel_capacity_sats.unwrap(), expected_channel_capacity);
```